### PR TITLE
feat(migrate): env var to disable advisory locking

### PR DESCRIPTION
**Pull request by Jolg42**
Closes https://github.com/prisma/prisma/issues/16020#issuecomment-1693451258
https://github.com/prisma/prisma/issues/19350

This adds a new environment variable `PRISMA_SCHEMA_DISABLE_ADVISORY_LOCK` to disable the advisory locking used by Migrate for `migrate deploy` or `migrate dev` for example.

Example values that disable the locking 
```
PRISMA_SCHEMA_DISABLE_ADVISORY_LOCK=1
PRISMA_SCHEMA_DISABLE_ADVISORY_LOCK=anything
```

Values that DO NOT disable the locking
```
PRISMA_SCHEMA_DISABLE_ADVISORY_LOCK=
PRISMA_SCHEMA_DISABLE_ADVISORY_LOCK=0
```

### CODE REVIEW

**Review by tomhoule — APPROVED** [review]

File: schema-engine/connectors/sql-schema-connector/src/lib.rs [hunk 1]
```diff
@@ -172,6 +172,21 @@ impl SchemaConnector for SqlSchemaConnector {
     }
 
     fn acquire_lock(&mut self) -> BoxFuture<'_, ConnectorResult<()>> {
+        // If the env is set and non empty or set to `0`, we disable the lock.
+        let disable_lock: bool = std::env::var("PRISMA_SCHEMA_DISABLE_ADVISORY_LOCK")
+            .ok()
+            .and_then(|value: String| match &value as &str {
+                "0" | "" => Some(false),
+                _ => Some(true),
+            })
```
Comment (aqrln) [inline schema-engine/connectors/sql-schema-connector/src/lib.rs:181]:
```suggestion
            .map(|value: String| match value.as_str() {
                "0" | "" => false,
                _ => true,
            })
```

**Comment by janpio** [comment]
Just for my mental picture: Does this affect all databases or only some?

**Comment by Jolg42** [comment]
All databases where we implement an advisory lock, so MySQL, PostgreSQL, SQLite, and SQL Server (nothing implemented for MongoDB)

**Comment by Jolg42** [comment]
Note: This is now fixed in `5.3.0-dev.13` if you want to try it out. Let us know if it works like expected.
It will be released in the official 5.3.0 version on Tuesday, September, 12th.

**Pull request merged**